src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -207,7 +207,7 @@ protected override void Free()
             ArrayBuilder<(BoundReturnStatement, TypeWithAnnotations)> returnTypesOpt,
             VariableState initialState,
             Action<BoundExpression, TypeWithAnnotations> callbackOpt)
-            : base(compilation, method, node, new EmptyStructTypeCache(compilation, dev12CompilerCompatibility: false), trackUnassignments: true)
+            : base(compilation, method, node, new EmptyStructTypeCache(compilation: null, false), trackUnassignments: true)
```

Comment:
>null [](start = 84, length = 4)

Nit, unrelated directly this PR: We might want to consider refactoring `EmptyStructTypeCache` so that the constructor is either empty, which means `dev12CompilerComatibility` is `false`, or takes a `Compilation`, which means that it's true. As it is, looking at this change it was very confusing what was going on here.

Reply:
I went through a couple of iterations of what kind of cache to use, finally ending back here.  I think static factories would be better than constructors.